src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -499,7 +499,7 @@ def load_arguments(self, command=None):
             command_loaders = self.cmd_to_loader_map.get(command, None)
 
         if command_loaders:
-            for loader in command_loaders:
+            for loader in sorted(command_loaders, key=lambda loader: (loader.__module__, loader.__class__.__name__)):
```

Comment:
For a command module, such as `role`, 

- `loader.__module__` evaluates to `azure.cli.command_modules.role`
- `loader.__class__.__name__` evaluates to `RoleCommandsLoader`

For an extension, such as `ad`,

- `loader.__module__` evaluates to `azext_ad`
- `loader.__class__.__name__` evaluates to `DomainServicesResourceProviderCommandsLoader`

The sorted list will look like

```
azext_a
azext_b
...
azext_z
azure.cli.command_modules.a
azure.cli.command_modules.b
...
azure.cli.command_modules.z
```

In my opinion, prioritizing extensions over command modules by string comparing `loader.__module__` (`azext_ad` < `azure.cli.command_modules.role`) is a little bit tricky. It is only a coincidence that `azext_` appears before `azure.` Things can get complex if we want to prioritize command modules.

Even if we want to prioritize extensions, it's better to do it programmatically, by either:

1. Prefixing the key tuple with another `0 if loader.__module__.startswith('azext') else 1` representing if `loader.__module__` is an extension and its priority.
2. Extracting extensions and command modules into 2 separate lists, sorting them separately and combining these 2 lists into one.